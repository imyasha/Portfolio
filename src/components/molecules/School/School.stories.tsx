import { mockSchool } from 'MockData';
import { School, SchoolProps } from './School';

export default {
	title: 'molecules/School',
	component: School,
};

export const SchoolStory = (args: SchoolProps) => <School {...args} />;

SchoolStory.storyName = 'School';
SchoolStory.args = mockSchool;
