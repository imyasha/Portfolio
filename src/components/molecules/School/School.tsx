import format from 'date-fns/format';
import { ICompany } from '@types';
import Image from 'next/image';

import { Box } from 'Atoms/Box';

export interface SchoolProps {
	institute: ICompany;
	course: string;
	date: string;
	technologies: string[];
}

const School = ({ institute, course, date, technologies }: SchoolProps) => {
	return (
		<Box>
			<div className="flex gap-4 items-center">
				{institute.logo && (
					<Image
						src={institute.logo}
						alt={institute.name}
						width={64}
						height={64}
						className="rounded-md"
					/>
				)}
				<div className="w-full mb-2">
					<div className="flex justify-between">
						<h3 className="font-bold">{course}</h3>
						<span>{format(new Date(date), 'MMM yyyy')}</span>
					</div>
					<h4>
						{institute.url ? (
							<a
								href={institute.url}
								target="_blank"
								rel="noopener noreferrer"
								className="underlined relative"
							>
								{institute.name}
							</a>
						) : (
							<>{institute.name}</>
						)}
					</h4>
					<p className="text-sm">
						<strong>Course Contents: </strong>
						{technologies.join(', ')}
					</p>
				</div>
			</div>
		</Box>
	);
};

export { School };
