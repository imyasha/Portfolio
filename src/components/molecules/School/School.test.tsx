import { mockSchool } from 'MockData';
import { render } from '@testing-library/react';
import { School } from '../School';

describe('School', () => {
	it('renders correctly', () => {
		const { container } = render(<School {...mockSchool} />);
		expect(container).toMatchSnapshot();
	});
});
