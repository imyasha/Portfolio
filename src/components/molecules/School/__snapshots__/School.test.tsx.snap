// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`School renders correctly 1`] = `
<div>
  <div
    class="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all"
  >
    <div
      class="flex gap-4 items-center"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2764%27%20height=%2764%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Some Company"
          class="rounded-md"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="w-full mb-2"
      >
        <div
          class="flex justify-between"
        >
          <h3
            class="font-bold"
          >
            Some course
          </h3>
          <span>
            Jan 2020
          </span>
        </div>
        <h4>
          <a
            class="underlined relative"
            href="https://somecompany.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Some Company
          </a>
        </h4>
        <p
          class="text-sm"
        >
          <strong>
            Course Contents: 
          </strong>
          TypeScript, React
        </p>
      </div>
    </div>
  </div>
</div>
`;
